aws/resource_aws_autoscaling_group.go

```diff
@@ -834,7 +838,7 @@ func resourceAwsAutoscalingGroupDrain(d *schema.ResourceData, meta interface{})
 
 	// Next, wait for the autoscale group to drain
 	log.Printf("[DEBUG] Waiting for group to have zero instances")
-	return resource.Retry(10*time.Minute, func() *resource.RetryError {
+	return resource.Retry(d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
```

Comment:
Is there any reason to decrease the default timeout from 10 to 5 mins here?
We could either introduce custom draining timeout or set the default to 10mins, I think.